AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -2049,6 +2052,67 @@ - (BOOL)supportsRangeManagedInterfaceState
   return ASHierarchyStateIncludesRangeManaged(_hierarchyState);
 }
 
+- (id<ASTraitCollection>)traitCollection
+{
+  ASDN::MutexLocker l(_propertyLock);
+  if (_traitCollection == nil) {
+    _traitCollection = [[ASDisplayTraitCollection alloc] initWithTraits:_internalTraits];
```

Comment:
I can see wanting to pass around some type of context with the trait collection... in fact I see this type of requirement in the current Pinterest app ;). Is there some way we could have a custom trait collection class so that would support this. 

For example, instead of calling `ASDisplayTraitCollection alloc`, we have some idea of a custom trait collection class so the init would look like `[[[ASDisplayNode traitCollectionClass] alloc] initWithTraits:_internalTraits];` That would allow for a subclass of `ASDisplayTraitCollection` that has a context, though I'm not entirely sure where that context would get set. I don't really want to make an `ASDisplayNode` subclass just for that...